src/index.js

```diff
@@ -112,21 +115,26 @@ class WebSockets {
   }
 
   /**
-   * Takes a list of `Multiaddr`s and returns only valid Websockets addresses
+   * Takes a list of `Multiaddr`s and returns only valid Websockets addresses.
+   * By default, in a browser environment only DNS+WSS multiaddr is accepted,
+   * while in a Node.js environment DNS+{WS, WSS} multiaddrs are accepted.
+   *
    * @param {Multiaddr[]} multiaddrs
    * @returns {Multiaddr[]} Valid Websockets multiaddrs
    */
   filter (multiaddrs) {
     multiaddrs = Array.isArray(multiaddrs) ? multiaddrs : [multiaddrs]
 
-    return multiaddrs.filter((ma) => {
-      if (ma.protoCodes().includes(CODE_CIRCUIT)) {
-        return false
-      }
+    if (this._filter) {
+      return this._filter(multiaddrs)
+    }
 
-      return mafmt.WebSockets.matches(ma.decapsulateCode(CODE_P2P)) ||
-        mafmt.WebSocketsSecure.matches(ma.decapsulateCode(CODE_P2P))
-    })
+    // Browser
+    if (env.isBrowser || env.isWebWorker) {
+      return filters.dnsWss(multiaddrs)
+    }
+
+    return filters.dnsWsOrWss(multiaddrs)
```

Comment:
This can probably be `all` in Node, because a TCP dial on ws is valid. We don't need crypto because we use Noise.

We might need to double check that node doesn't block this though. I think our tests are only using loopback addresses right now.

Reply:
Added a non loopback addr test to a local addr